eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py

```diff
@@ -1,39 +1,69 @@
 import argparse
-import glob
 import os
 import pickle
-import re
-from pytz import timezone
 
 import psycopg2
+import psycopg2.extras
 from collections import defaultdict
 from datetime import datetime
 
 from ebi_eva_common_pyutils.logger import logging_config
 from ebi_eva_common_pyutils.mongodb import MongoDatabase
-from ebi_eva_common_pyutils.config_utils import get_pg_metadata_uri_for_eva_profile
+from ebi_eva_common_pyutils.config_utils import get_pg_metadata_uri_for_eva_profile, get_accession_pg_creds_for_profile
 from ebi_eva_common_pyutils.pg_utils import execute_query, get_all_results_for_query
-from gather_clustering_counts.gather_per_species_clustering_counts import assembly_table_name
-
+from gather_clustering_counts.gather_per_species_clustering_counts import assembly_table_name, tracker_table_name
+from urllib.parse import urlsplit
 
 logger = logging_config.get_logger(__name__)
 logging_config.add_stdout_handler()
 taxonomy_scientific_name_map = {}
+assembly_taxonomy_map = defaultdict(set)
 
 
-def gather_count_from_mongo(clustering_dir, remapping_dir, mongo_source, private_config_xml_file, release_version):
-    # Assume the directory structure:
-    # clustering_dir --> <scientific_name_taxonomy_id> --> <assembly_accession> --> cluster_<date>.log_dict
-    all_clustering_logs_pattern = os.path.join(clustering_dir, '*', 'GCA_*', 'cluster_*.log')
-    # Beginning release 4, clustering of new studies is also carried out during remapping
-    all_remapping_logs_pattern = os.path.join(remapping_dir, '*', 'GCA_*', 'logs', '*_clustering.log')
-    all_log_files = glob.glob(all_clustering_logs_pattern) + glob.glob(all_remapping_logs_pattern)
+def get_release_end_timestamps_from_last_release(private_config_xml_file, current_release_version) -> dict:
+    """
+    Get release end timestamps from last release for all assemblies
+    """
+    assembly_timestamp_map = {}
+    with psycopg2.connect(get_pg_metadata_uri_for_eva_profile("production_processing", private_config_xml_file),
+                          user="evapro") as metadata_connection_handle:
```

Comment:
```suggestion
    with get_metadata_connection_handle("production_processing", private_config_xml_file) as metadata_connection_handle:
```
and similar elsewhere

Reply:
Addressed